content/docs/command-reference/run.md

```diff
@@ -244,14 +244,10 @@ $ dvc run -n my_stage './my_script.sh $MYENVVAR'
   command's code is non-deterministic
   ([not recommended](#avoiding-unexpected-behavior)).
 
-- `--no-commit` - do not save outputs to cache. A stage created, while nothing
-  is added to the cache. In the stage file, the file hash values will be empty;
-  They will be populated the next time this stage is actually executed, or
-  `dvc commit` can be used to force committing existing output file versions to
-  cache.
-
-  This is useful to avoid caching unnecessary data repeatedly when running
-  multiple experiments.
+- `--no-commit` - do not store the outputs of this execution in the cache
+  (`dvc.yaml` and `dvc.lock` are still created or updated); useful to avoid
+  caching unnecessary data when executing tests or experiments. Use `dvc commit`
```

Comment:
tests and experiments ...

both are overloaded ... 

may be something like `when iterations on code and data happen fast`? 

also in this specific case - it's about creating the pipeline first ... (then run it as a whole)

Reply:
Yeah in the `commit` ref. this is now explained as "when exploring different data or stages". Let me think 🧠

Reply:
> iterations on code and data happen fast

But why do people iterate fast? It's during an exploration/testing/experimenting phase of the development process. I guess "exploring" is the only non-overloaded term there. Other phrases: spatial search, brainstorm, play around

Reply:
> it's about creating the pipeline first

That applies more to --no-exec I think, let me check it's mentioned clearly... ⌛

Reply:
> applies more to --no-exec I think, let me check it's mentioned clearly

Wait no, we already discussed this in a recent PR over the import refs. and decided to remove the motivation paragraph because the option description was too long, among other reasons...

Reply:
It is, however, mentioned in `run` already, as "This is useful if, for example, you need to build a pipeline quickly first, and run it all at once later (with `dvc repro`)." so... I guess we should add it to imports? Added for now! (in